pkg/common/utils/utils.go

```diff
@@ -240,18 +240,21 @@ func MergeMaps(first, second map[string]string) map[string]string {
 // find user name from pod label
 func GetUserFromPod(pod *v1.Pod) string {
 	userLabelKey := conf.GetSchedulerConf().UserLabelKey
-	// User name to be defined in labels
-	for name, value := range pod.Labels {
-		if name == userLabelKey {
-			log.Logger().Info("Found user name from pod labels.",
-				zap.String("userLabel", userLabelKey), zap.String("user", value))
-			return value
-		}
+	// UserLabelKey should not be empty
+	if len(userLabelKey) == 0 {
+		userLabelKey = constants.DefaultUserLabel
+		log.Logger().Debug("userLabelKey is empty and replaced by default setting",
+			zap.String("userLabel", userLabelKey))
 	}
-	value := constants.DefaultUser
 
+	// User name to be defined in labels
+	if username, ok := pod.Labels[userLabelKey]; ok {
```

Comment:
Can the label be found and still return an empty username?
We need to handle that case and still set the default user

Reply:
Yes, the default user name will be return if yunikorn can't get the username from the pod.
I added the modification in the current commit.

Reply:
A label value can be empty. That would result in `ok` to be set to true but username to be "" (or empty string) in that case we should assign the default user also.
That is not covered by the change.

Reply:
Ok, I am going to add relevant modifications for this case.